### PR TITLE
Add Chinese monoid presentation

### PR DESCRIPTION
**Pull request by MTWhyte**
This PR adds a presentation construction for the [Chinese monoid](https://en.wikipedia.org/wiki/Chinese_monoid).

### CODE REVIEW

File: tests/test-knuth-bendix-6.cpp [hunk 1]
```diff
@@ -35,11 +35,12 @@
 //
 // 6: contains tests for congruence::KnuthBendix.
 
-// #define CATCH_CONFIG_ENABLE_PAIR_STRINGMAKER
```
Comment (james-d-mitchell) [inline tests/test-knuth-bendix-6.cpp:38]: Please leave this line commented out, it's not actually used for anything.

File: tests/fpsemi-examples.cpp [hunk 2]
```diff
@@ -24,7 +24,7 @@
 #include <algorithm>  // for max, for_each
 #include <cstdint>    // for int64_t
 #include <cstdlib>    // for abs
-#include <numeric>    // for iota
+#include <numeric>  // for iota
```
Comment (james-d-mitchell) [inline tests/fpsemi-examples.cpp:27]: Please run `make format` and push again! I'll merge after that, nice one @MTWhyte

**Comment by james-d-mitchell** [comment]
BTW the `libsemigroups_pybind11` CI job will fail because of changes to `presentation::sort_rules` in `master`.

**Comment by hivert** [comment]
Hi there. I've been away of my research repo for a long time and just connected back. I was surprised to see the chinese monoid. Glad it made its ways there :-)

**Pull request merged**